src/plugins/pass/main.js

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/gjs
+
+const { GLib, Gio, Notify } = imports.gi;
+
+Notify.init('Password Store');
+
+/** The directory that this script is executed from. */
+const SCRIPT_DIR = GLib.path_get_dirname(new Error().stack.split(':')[0].slice(1));
+
+/** Add our directory so we can import modules from it. */
+imports.searchPath.push(SCRIPT_DIR)
+
+const STDIN = new Gio.DataInputStream({ base_stream: new Gio.UnixInputStream({ fd: 0 }) })
+const STDOUT = new Gio.DataOutputStream({ base_stream: new Gio.UnixOutputStream({ fd: 1 }) })
+
+
+function list_passwords(root, directory) {
+    const entries = directory.enumerate_children(
+        Gio.FILE_ATTRIBUTE_STANDARD_NAME + "," +
+        Gio.FILE_ATTRIBUTE_STANDARD_TYPE + "," +
+        Gio.FILE_ATTRIBUTE_STANDARD_IS_HIDDEN,
+        Gio.FileQueryInfoFlags.NOFOLLOW_SYMLINKS,
+        null
+    );
+
+    let entry;
+    let results = [];
+
+    while((entry = entries.next_file(null)) !== null) {
+        const name = entry.get_name();
+        const target_file = directory.get_child(entry.get_name());
+
+        if (entry.get_file_type() === Gio.FileType.DIRECTORY) {
+            results = results.concat(list_passwords(root, target_file))
+        } else if (name.endsWith('.gpg')) {
+            let path = root.get_relative_path(target_file);
+            results.push(path.replace('.gpg', ''));
+        }
+    }
+    return results;
+}
+
+function send_notification(body, icon) {
+    let notification = Notify.Notification.new(
+        'Password Store',
+        body,
+        icon,
+    );
+    notification.show();
+}
+
+
+class App {
+    constructor() {
+        let home_dir = GLib.get_home_dir();
+        let password_store = Gio.File.new_for_path(`${home_dir}/.password-store`);
+
+        if (password_store.query_exists(null)) {
+            this.passwords = list_passwords(password_store, password_store);
+        } else {
+            this.passwords = [];
+        }
+
+        this.selections = [];
+
+        let i = 0;
+        for (i = 0; i < this.passwords.length; i++) {
+            let password = this.passwords[i];
+            this.selections.push({
+                id: i,
+                name: password,
+                description: null,
+                icon: 'dialog-password',
+            })
+        }
```

Comment:
Instead of doing manual iteration over a collection of items, it might be better to let the compiler figure it out :)
Also, this way, you get both the item and its index for free, as they are the 1st and 2nd args in `Array.prototype.forEach()`. You can also name them however you want, including ways that can take advantage of object shorthand notation for a little bit of code golf
```
this.passwords.forEach((name, id) => {
            this.selections.push({
                    id,
                    name,
                    description: null,
                    icon: 'dialog-password',
})
```